importers/aircraft_csv_format_importer.py

```diff
@@ -0,0 +1,155 @@
+from datetime import datetime
+
+from pepys_import.core.formats import unit_registry
+from pepys_import.core.formats.location import Location
+from pepys_import.core.validators import constants
+from pepys_import.file.highlighter.support.combine import combine_tokens
+from pepys_import.file.importer import Importer
+from pepys_import.utils.sqlalchemy_utils import get_lowest_privacy
+from pepys_import.utils.unit_utils import convert_absolute_angle, convert_distance, convert_speed
+
+
+class AircraftCsvFormatImporter(Importer):
+    def __init__(self):
+        super().__init__(
+            name="Aircraft CSV Format Importer",
+            validation_level=constants.BASIC_LEVEL,
+            short_name=" Aircraft CSV Importer",
+            default_privacy="Public",
+            datafile_type="CSV",
+        )
+        self.text_label = None
+
+    def can_load_this_type(self, suffix):
+        return suffix.upper() == ".CSV"
+
+    def can_load_this_filename(self, filename):
+        return True
+
+    def can_load_this_header(self, header):
+        return header.startswith("Date(Uk)")
+
+    def can_load_this_file(self, file_contents):
+        return True
+
+    def _load_this_line(self, data_store, line_number, line, datafile, change_id):
+        # Skip the header
+        if line_number == 1:
+            return
+
+        tokens = line.tokens(line.CSV_TOKENISER, ",")
+        if len(tokens) <= 1:
+            # The last line may be empty, don't worry
+            return
+        if len(tokens) < 8:
+            # CSV example file has 9 columns
+            self.errors.append(
+                {self.error_type: f"Error on line {line_number}. Not enough tokens: {line}"}
+            )
+            return
+
+        # Separate token strings
+        date_token = tokens[0]
+        time_token = tokens[1]
+        lat_degrees_token = tokens[4]
+        long_degrees_token = tokens[5]
+        altitude_token = tokens[6]
+        speed_token = tokens[7]
+        course_token = tokens[8]
+
+        # Check Date is in correct format
+        if len(date_token.text) != 10:
+            self.errors.append(
+                {
+                    self.error_type: f"Error on line {line_number}. Date format '{date_token.text}' "
+                    f"should be 10 figure data"
+                }
+            )
+            return
+
+        # Times always in Zulu/GMT
+        if len(time_token.text) != 8:
+            self.errors.append(
+                {
+                    self.error_type: f"Line {line_number}. Error in Time format '{time_token.text}'."
+                    "Should be HH:mm:ss"
+                }
+            )
+            return
+
+        timestamp = self.parse_timestamp(date_token.text, time_token.text)
+        if timestamp:
+            combine_tokens(date_token, time_token).record(self.name, "timestamp", timestamp)
+        else:
+            # Skip line if invalid timestamp
+            self.errors.append(
```

Comment:
You seem to be missing a test for a timestamp that doesn't parse correctly - so this if isn't being covered by tests. I think you've got a test for the time being the wrong length (ie. not 8 char time), but not for an invalid date/time like 01:99:00 or 99/03/2015 etc)